### PR TITLE
Pie chart component

### PR DESCRIPTION
**Pull request by jcc37**
* Add `PieChartD3` and `PieChartComponent`
* Add tests for new components and D3 classes

### CODE REVIEW

File: src/datasets/summarizations/category-topk-coverage.summarization.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import * as math from 'mathjs';
+import { Summary } from './types';
+import { CategoricalPoint } from '../metas/types';
+import { cacheSummaries, createOrdinalText } from './utils/commons';
+import { formatY } from '../../utils/formatters';
+
+export interface Config {
+  topk: number;
```
Comment (Rhombus33) [inline src/datasets/summarizations/category-topk-coverage.summarization.ts:8]: nit: camelCase

File: src/datasets/summarizations/category-topk-coverage.summarization.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import * as math from 'mathjs';
+import { Summary } from './types';
+import { CategoricalPoint } from '../metas/types';
+import { cacheSummaries, createOrdinalText } from './utils/commons';
+import { formatY } from '../../utils/formatters';
+
+export interface Config {
```
Comment (Rhombus33) [inline src/datasets/summarizations/category-topk-coverage.summarization.ts:7]: should the configs for different summaries have more specific names to differentiate them?

File: src/datasets/summarizations/category-topk.summarization.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { Summary } from './types';
+import { CategoricalPoint } from '../metas/types';
+import { cacheSummaries, createOrdinalText } from './utils/commons';
+
+export interface Config {
+  topk: number;
```
Comment (Rhombus33) [inline src/datasets/summarizations/category-topk.summarization.ts:6]: nit: camelCase?

**Review by Rhombus33 — APPROVED** [review]

File: src/datasets/summarizations/category-topk.summarization.ts (hunk 3 above)
Comment (jcc37) [reply src/datasets/summarizations/category-topk.summarization.ts:6]: Isn't "topk" a word?

File: src/datasets/summarizations/category-topk-coverage.summarization.ts (hunk 2 above)
Comment (jcc37) [reply src/datasets/summarizations/category-topk-coverage.summarization.ts:7]: That is an intended design. I want the some of the config to be shared between multiple summarization modules. Take `topk` as an example, if you are curious about the the top 3 item summaries, then the coverage of these 3 items can be a follow up to those summaries. This design (shared configs) is widely used in refactor in common configs like "metric" and "metricUnit".

File: src/datasets/summarizations/category-bucket-comparison.summarization.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    }];
+  });
+}
+
+function bucketizePoints(points: CategoricalPoint[], bucketPercentageTolerance: number) {
+  const buckets: CategoricalPoint[][] = [];
+  let currentBucket: CategoricalPoint[] = [];
+
+  for (const { x, y } of points) {
+    const currentBucketYMax = currentBucket[0]?.y ?? null;
```
[the remark this answers is not in the document]
Comment (jcc37) [reply src/datasets/summarizations/category-bucket-comparison.summarization.ts:61]: OK

File: src/d3/pie-chart.d3.spec.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import { PieChartD3 } from './pie-chart.d3';
+import { CategoricalPoint } from '../datasets/metas/types';
+import { PieChartDatum } from '../components/pie-chart/pie-chart.component';
+import { RenderOptions } from './xy-chart.d3';
+
+interface SubjectRenderOptions extends RenderOptions<CategoricalPoint, PieChartDatum> {
+  data$: Subject<PieChartDatum[]>;
+}
+
+const mockData = [...Array(10).keys()].map(datumIndex => ({
```
[the remark this answers is not in the document]
Comment (jcc37) [reply src/d3/pie-chart.d3.spec.ts:12]: Every time when I want to add some test data, I always go to `utils/mock.spec.ts` and copy & paste the code that randomly generates mock data. Keep forgetting we should use hardcoded mock data.

Fixed

**Pull request merged**